frontend/src/components/editionOverview.tsx

```diff
@@ -0,0 +1,229 @@
+import { Col, Container, Row, Toast, ToastContainer } from "react-bootstrap";
+import styles from "../styles/editionOverview.module.css";
+import { capitalize } from "../utility/stringUtil";
+import Image from "next/image";
+import { useState } from "react";
+import useTranslation from "next-translate/useTranslation";
+import useSWR, { useSWRConfig } from "swr";
+import apiPaths from "../properties/apiPaths";
+import { getEditionOnUrl } from "../api/calls/editionCalls";
+import { AxiosResponse } from "axios";
+import { StatusCodes } from "http-status-codes";
+import {
+    editionSaveActiveHandler,
+    editionSaveNameHandler,
+    editionSaveYearHandler,
+} from "../handlers/editionHandler";
+import timers from "../properties/timers";
+
+export function EditionOverview(editionId: any) {
+    const { t } = useTranslation("common");
+    const { mutate } = useSWRConfig();
+
+    const [name, setName] = useState<string>("");
+    const [editName, setEditName] = useState<boolean>(false);
+    const [year, setYear] = useState<string>("");
+    const [editYear, setEditYear] = useState<boolean>(false);
+    const [active, setActive] = useState<boolean>(false);
+    const [showGeneralError, setShowGeneralError] = useState<boolean>(false);
+    const [showYearError, setShowYearError] = useState<boolean>(false);
+
+    let { data, error } = useSWR(apiPaths.editions + "/" + editionId.editionId, getEditionOnUrl);
+
+    if (error || !data) {
+        return null;
+    }
+
+    function handleEditName() {
+        if (data) {
```

Comment:
this check is not needed right? The edition is never undefined if you call this?

Reply:
(multiple times in this file)

Reply:
True, yet TypeScript still throws errors if I remove them, so I guess I will let them stay